main.py

```diff
@@ -629,27 +722,19 @@ def handle_link_liveness_disabled(self, interfaces) -> None:
         for link in links.values():
             self.notify_link_status_change(link, reason="liveness_disabled")
 
-    @listen_to("kytos/.*.link_available_tags")
-    def on_link_available_tags(self, event):
-        """Handle on_link_available_tags."""
-        with self._links_lock:
-            self.handle_on_link_available_tags(event.content.get("link"))
-
-    def handle_on_link_available_tags(self, link):
-        """Handle on_link_available_tags."""
-        if link.id not in self.links:
-            return
-        endpoint_a = self.links[link.id].endpoint_a
-        endpoint_b = self.links[link.id].endpoint_b
-        values_a = [tag.value for tag in endpoint_a.available_tags]
-        values_b = [tag.value for tag in endpoint_b.available_tags]
-        ids_details = [
-            (endpoint_a.id, {"_id": endpoint_a.id,
-                             "available_vlans": values_a}),
-            (endpoint_b.id, {"_id": endpoint_b.id,
-                             "available_vlans": values_b})
-        ]
-        self.topo_controller.bulk_upsert_interface_details(ids_details)
+    @listen_to("kytos/core.interface_tags")
+    def on_interface_tags(self, event):
+        """Handle on_interface_tags."""
+        with self._interface_lock:
```

Comment:
Nicely done how you reused this event not only from kytos-core as a client, but also here via the REST endpoints when it needs to persist the changes. Still, there two things to be addressed here:

1 - self._interface_lock as a `Lock` instance currently should work as expected, but you can increase the concurrency throughput here if you were to use a `defaultdict(Lock)` just so this is only locked per interface, since MongoDB ensures thread safety per document then it should be OK. So if you could adapt accordingly that'd be great. 
2 - Notice that you can still have out of order events here, for instance an older event that immediately gets supended before getting the lock, another thread executes, and then the old one tries to execute, since `interface` is a shared object and protected with locks theoretically the contents should be the same, but still, it can hit unnecessarily the DB in those cases. Since this part is very critical and is dealing with core resources, to be extra safe I think you should add a similar check with an extra dict inside the lock keeping track of when the interface was last updated, and if it's older event discard it, similar [to how we do on this part, check it out](https://github.com/kytos-ng/topology/blob/master/main.py#L733-L739)

Reply:
Checking the order of events now.